test/integration/confirmations/signatures/personalSign.test.tsx

```diff
@@ -0,0 +1,259 @@
+import { fireEvent, waitFor } from '@testing-library/react';
+import { ApprovalType } from '@metamask/controller-utils';
+import mockMetaMaskState from '../../data/integration-init-state.json';
+import { integrationTestRender } from '../../../lib/render-helpers';
+import { shortenAddress } from '../../../../ui/helpers/utils/util';
+import * as backgroundConnection from '../../../../ui/store/background-connection';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+  MetaMetricsEventLocation,
+} from '../../../../shared/constants/metametrics';
+
+jest.mock('../../../../ui/store/background-connection', () => ({
+  ...jest.requireActual('../../../../ui/store/background-connection'),
+  submitRequestToBackground: jest.fn(),
+}));
+
+const mockedBakcgroundConnection = backgroundConnection as jest.Mocked<
+  typeof backgroundConnection
+>;
+
+const backgroundConnectionMocked = {
+  onNotification: jest.fn(),
+};
+
+describe('PersonalSign', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.resetAllMocks();
+  });
+
+  afterAll(() => {
+    jest.clearAllMocks();
+    jest.resetAllMocks();
+  });
+
+  it('displays the header account modal with correct data', async () => {
+    const account =
+      mockMetaMaskState.internalAccounts.accounts[
+        mockMetaMaskState.internalAccounts
+          .selectedAccount as 'cf8dace4-9439-4bd4-b3a8-88c821c8fcb3'
+      ];
+
+    const accountName = account.metadata.name;
+    const pendingPersonalSignTime = new Date().getTime();
+    const pendingPersonalSignId = '0050d5b0-c023-11ee-a0cb-3390a510a0ab';
+    const mockedMetaMaskState = {
```

Comment:
There's alot of state setup in these tests and it appears lots of it is duplicated. Should we define some constants for this test specific state that we can copy and mutate? Or alternatively a helper function such as `buildState` with property arguments for the state that varies between tests?